### PR TITLE
add support for defaults - fixes #3

### LINKED ISSUE #3: Defaults not always correct
[this](https://github.com/editorconfig/editorconfig-vscode/blob/master/src/editorConfigMain.ts#L118-L120) block tries to use the defaults of the texteditor and in my case, in most circumstances, it works. But not always. Let's assume we have this `.editorconfig` file.

```ini
root = true

[{package.json,*.yml}]
indent_style = space
```

### Expected behaviour
If my VS Code settings of `insertSpaces` is set to `auto`, it should auto-detect the the indentation type (spaces or tabs). This means that if I open `index.js`, (which uses tabs) it should detect the `tabs` and use that is indent style.

### Current behaviour

1. Open a project and open `package.json` as the FIRST file ([this](https://github.com/editorconfig/editorconfig-vscode/blob/master/src/editorConfigMain.ts#L118-L120) will set `spaces` as default indent type)
2. Open `index.js` (which uses tabs) and it will use spaces as current indent type

I will create an issue in the VSCode issue tracker and link this one. They have to refactor the API in order for us to support automatically detected defaults by VS Code.

### PR DESCRIPTION
**Pull request by SamVerschueren**
I managed to find a solution for #3  :). Wonder why I didn't come up with that in the first place.

So `textEditor.options` is of type `TextEditorOptions` which is implemented like this

```ts
interface TextEditorOptions {
    tabSize: number,
    insertSpaces: boolean
}
```

But actually, it is possible to provide the string `auto` and then it'll work. So that is what I do now. I read the global workspace settings of the user (can be `auto`, can be a number or a boolean). And I use those as defaults. If the global configuration changes, an event is fired and the plugin reads the new settings. So if my global settings look like

```json
{
    "editor.insertSpaces": "auto",
    "editor.tabSize": "auto"
}
```

And my `.editorconfig` looks like

```ini
[*]
indent_style = tab

[{package.json,*.yml}]
indent_style = space
indent_size = 2
```

When I open `index.js`, the applied options object will be

```js
textEditor.options = {
    insertSpaces: false,
    tabSize: 'auto'
};
```

And the tab size is auto calculated :).

### CODE REVIEW

**Comment by SamVerschueren** [comment]
Not sure why the tests fail though. Any idea @jedmao ?

File: src/editorConfigMain.ts [hunk 1]
```diff
@@ -48,6 +53,11 @@ class DocumentWatcher implements IEditorConfigProvider {
 			}
 		}));
 
+		// Listen for changes in the configuration
+		subscriptions.push(workspace.onDidChangeConfiguration(() => {
```
Comment (jedmao) [inline src/editorConfigMain.ts]: Instead of a lambda function here, can't you do:
```ts
this._onConfigChanged.bind(this)
```

**Comment by jedmao** [comment]
Looks like you figured out how to fix the tests! I'll try this out locally later tonight if I can find the time. Thanks for the PR!

File: src/editorConfigMain.ts (hunk 1 above)
Comment (SamVerschueren) [inline src/editorConfigMain.ts]: Might be cleaner indeed.

**Comment by SamVerschueren** [comment]
I think it was a Travis issue, didn't do anything :).

**Comment by SamVerschueren** [comment]
@jedmao Changed the binding of the onConfigChanged function.

**Comment by jedmao** [comment]
The Travis error is:
> Error installing vscode.d.ts: TypeError: Object #<Object> has no method 'map'
I've seen this error before, but I didn't know it was intermittent. Will restart the build and see what happens.

**Comment by SamVerschueren** [comment]
That was the one I had before.

**Comment by SamVerschueren** [comment]
NICE! Thanks.

// Offtopic
Looks like you're up late :)
screen shot 2016-02-12 at 09 28 40

**Comment by jedmao** [comment]
@SamVerschueren you are correct. It seems the only time I can get any actual code written is between 2 and 5 AM. Daytime is so distracting!

**Comment by SamVerschueren** [comment]
Hahaha :D. That fits better in my timezone then ;).

**Comment by jedmao** [comment]
Published as [v0.2.2](https://github.com/editorconfig/editorconfig-vscode/commit/1bd7de386e03bfcb1ccc706e86c4cde004089e99)!

**Comment by SamVerschueren** [comment]
Awesome, thank you very much! This makes `.editorconfig` almost perfect in VS Code :).

**Comment by jedmao** [comment]
Mostly, thanks to you!

**Comment by jedmao** [comment]
Cool [`dev-time-cli`](https://www.npmjs.com/package/dev-time-cli) tool, BTW (I see that you're the author). I installed it and gave it a shot! Your name is a bit harder to type than mine, however :)

**Comment by SamVerschueren** [comment]
Thanks :).

> Your name is a bit harder to type than mine, however :)

True :D

**Pull request merged**